### ISSUE TITLE
Every wall after the first hand is a global constant: `_init` never seeds `round_state.rng_key`

### ISSUE THREAD

**Issue by Jannchie** [nissymori/mahjax#71]
### Summary

In both `red_mahjong` and `no_red_mahjong`, `_init` splits a `subkey` off the
init rng but never stores it — `_make_state` is called without `rng_key=`, so
`round_state.rng_key` keeps the `State` dataclass default `jax.random.PRNGKey(0)`.
Since `Env.step()` ignores its `key` argument (`del key`) and every reshuffle
derives from `round_state.rng_key` alone (`_prepare_next_round_assets`), **every
wall after the first hand is drawn from one global chain shared by all seeds,
envs, and runs**. The second hand's deck, all four deals, and the dora
indicator are literal constants of the library.

(The no_red `_init` docstring even lists "rng key" among the variables it sets —
the split was clearly intended to be stored.)

### Repro

```python
import jax, jax.numpy as jnp, numpy as np, mahjax

env = mahjax.make("red_mahjong", round_mode="half", next_round_style="dummy_share")

def second_wall(seed):
    state = env.init(jax.random.PRNGKey(seed))
    step, rng, prev = jax.jit(env.step), jax.random.PRNGKey(seed + 1), False
    while True:
        rng, k1, k2 = jax.random.split(rng, 3)
        act = jax.random.categorical(k1, jnp.where(state.legal_action_mask, 0.0, -1e9))
        state = step(state, act.astype(jnp.int32), k2)
        term = bool(state.round_state.terminated_round)
        if prev and not term:
            return np.asarray(state.round_state.deck)
        prev = term

print(np.array_equal(second_wall(0), second_wall(424242)))  # True (!)
```

### Why this matters for RL

From hand 2 on, the full wall — including the other three players' deals — is
identifiable from public observations (`round`, `honba`, and the dora
indicator, which is a constant per hand index). A self-play policy can
therefore memorize walls, which acts as an implicit full-information oracle:
it inflates in-simulator strength and evaluation numbers, and none of it
transfers to opponents/evaluations with properly randomized walls. It also
collapses the effective wall diversity of any multi-env rollout (all envs
share one wall sequence after their first hand).

### Fix

Pass the already-split (currently dead) `subkey` into `_make_state` as
`rng_key=` in both envs. The first hand's deal and dealer are bitwise
unchanged (they only ever used the other half of the split), so same-seed
reproducibility is preserved. Draft PR incoming.
